app/src/main/java/com/rajatgoyal/puzzle15/ui/GameActivity.java

```diff
@@ -330,22 +181,17 @@ protected void onPause() {
     protected void onResume() {
         super.onResume();
         if (!gameOver) {
-            resumeTimer();
+            startTimer(prevTimeMillis);
         }
     }
 
     @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-
-        outState.putIntArray("matrix_row_0", m[0]);
-        outState.putIntArray("matrix_row_1", m[1]);
-        outState.putIntArray("matrix_row_2", m[2]);
-        outState.putIntArray("matrix_row_3", m[3]);
-
-        outState.putInt("moves", moves);
-
-        outState.putLong("currTime", currTime);
+    protected void onStop() {
+        super.onStop();
+        Timber.d("on stop");
+        SharedPref.setGameMatrix(gameMatrix);
+        SharedPref.setMoves(moves);
+        SharedPref.setGameTime(currTime.toMillis());
```

Comment:
Save matrix only when `gameOver` variable is set to false. This will solve the issue in which there appears the resume option even after completing the game.